lib/spark.js

```diff
@@ -418,3 +494,64 @@ function mapErrorToStatusCode(err) {
   // Fallback code for unknown errors
   return 600
 }
+
+/**
+ * Picks a random provider based on their priority and supplied randomness.
+ *
+ * Providers are prioritized in the following order:
+ *
+ * 1. HTTP Providers with Piece Info advertised in their ContextID.
+ * 2. Graphsync Providers with Piece Info advertised in their ContextID.
+ * 3. HTTP Providers.
+ * 4. Graphsync Providers.
+ *
+ * @param {Provider[]} providers
+ * @param {number} randomness
+ * @returns {Provider | undefined}
+ */
+export function pickRandomProvider(providers, randomness) {
+  const rng = new Prando(randomness)
+
+  const filterByProtocol = (items, protocol) =>
+    items.filter((provider) => provider.protocol === protocol)
+
+  const pickRandomItem = (items) => {
+    if (!items.length) return undefined
+    return items[Math.floor(rng.next() * items.length)]
```

Comment:
IIUC, we are making exactly one `rng.next()` call per each randomness value. Using a pseudo-random generator for that feels like unnecessary complexity to me.

Can you treat the DRAND randomness as the random value instead?

Something along the following lines:

```js
// Take the first 16 hex characters and parse them as an integer
const randomValue = BigInt("0x" + randomness.slice(16))
// 16 characters, each character represents one of 16 values
const max = 16n**16n
const ix = Number(BigInt(items.length) * randomValue / max)
return items[ix]
```

Reply:
Great suggestion, I like the simplicity of it.

At first I tried implementing my own psuedo-RNG but leaned towards using `prando` as it was somewhat popular implementation.

I agree that this is much simpler and does not come with overhead of `prando`.